### PR TITLE
Context for aria-label on cart quantity controls

### LINKED ISSUE #7492: Improving aria-label usage based on context
While doing a pair-programming session on an unrelated task, I was left wondering how good our use of aria-label for accessibility purposes is and how it can be improved.

Best way to demonstrate this is with a quick example.

The “Remove item” action in the Cart block currently has no aria-label value; using Voice Over, it reads out only as the text it has. The shortcode version uses an “X” as the button text and has an aria-label value of “Remove this item.”. Both possess no labels. This is a bit generic without context. A user might not be sure what item is being removed from the cart.

I took a look at Amazon, for example, on the same functionality, and they provide context:

image

Apple.com does it via hidden text:

image

farfetch.com

image

(I also tried nike.com, but their cart was broken due to a CORS issue. Oopsie 😂)

Since we probably have multiple places where providing context would be an enhancement, a [good suggestion](p1664379675072159/1664378419.059999-slack-C8X6Q7XQU) about what could be done to improve on this was given on slack by @tarhi-saad:

> Maybe a similar effort in [Call to action to improve our public-facing text](https://wooblocksrepo.wordpress.com/2022/07/18/call-to-action-to-improve-our-public-facing-text/) can be made here as well! (e.g., create a document with a list of all aria-labels text, propose a more clear alternative, etc.)

### PR DESCRIPTION
**Pull request by wavvves**
The “Remove item” action in the Cart block currently has no aria-label value; using Voice Over, it reads out only as the text it has. This is a bit generic without context. A user might not be sure what item is being removed from the cart. The same goes for quantity controls.

Partially Fixes #7492

#### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Add some items to the cart
2. Using a screen reader, ensure it reads "increase/decrease quantity of <item>" / "remove <item> from your cart" on quantity controls, giving clear instructions on what item the amount on cart is being changed.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Added context for aria-label on cart quantity controls

### CODE REVIEW

File: assets/js/base/components/quantity-selector/index.tsx [hunk 1]
```diff
@@ -222,7 +230,7 @@ const QuantitySelector = ( {
 					onChange( newQuantity );
 					speak(
 						sprintf(
-							/* translators: %s refers to the item name in the cart. */
+							/* translators: %s refers to the item new quantity in the cart. */
```
Comment (opr) [inline assets/js/base/components/quantity-selector/index.tsx]:
```suggestion
							/* translators: %s refers to the item's new quantity in the cart. */
```

File: assets/js/base/components/quantity-selector/index.tsx [hunk 2]
```diff
@@ -211,9 +215,13 @@ const QuantitySelector = ( {
 				&#65293;
 			</button>
 			<button
-				aria-label={ __(
-					'Increase quantity',
-					'woo-gutenberg-products-block'
+				aria-label={ sprintf(
+					/* translators: %s refers to the item name in the cart. */
```
Comment (opr) [inline assets/js/base/components/quantity-selector/index.tsx]:
```suggestion
					/* translators: %s refers to the item's name in the cart. */
```

File: assets/js/base/components/quantity-selector/index.tsx [hunk 3]
```diff
@@ -197,7 +201,7 @@ const QuantitySelector = ( {
 					onChange( newQuantity );
 					speak(
 						sprintf(
-							/* translators: %s refers to the item name in the cart. */
+							/* translators: %s refers to the item new quantity in the cart. */
```
Comment (opr) [inline assets/js/base/components/quantity-selector/index.tsx]:
```suggestion
							/* translators: %s refers to the item's new quantity in the cart. */
```

File: assets/js/blocks/cart/cart-line-items-table/cart-line-item-row.tsx [hunk 4]
```diff
@@ -311,6 +311,14 @@ const CartLineItemRow: React.ForwardRefExoticComponent<
 							{ showRemoveItemLink && (
 								<button
 									className="wc-block-cart-item__remove-link"
+									aria-label={ sprintf(
+										/* translators: %s refers to the item name in the cart. */
```
Comment (opr) [inline assets/js/blocks/cart/cart-line-items-table/cart-line-item-row.tsx]:
```suggestion
										/* translators: %s refers to the item's name in the cart. */
```

**Review by opr — APPROVED** [review]
Works great! I just added some suggestions to improve comment grammar, nothing blocking so I will approve. Thanks for working on this!

**Comment by wavvves** [comment]
> Works great! I just added some suggestions to improve comment grammar, nothing blocking so I will approve. Thanks for working on this!

Thanks @opr! Committed the suggestions and I'm going to merge it after tests re-run 👍🏼

**Pull request merged**